artifacts.go

```diff
@@ -71,39 +70,51 @@ func read(aPath string) (parser.Workers, error) {
 	if ar == nil {
 		return nil, errors.New("Can not read artifact file.")
 	}
-	defer ar.Close()
 
 	p := parser.RootfsParser{
 		W: ioutil.Discard, // don't store update anywhere
 	}
 	ar.Register(&p)
 
-	return ar.Read()
+	_, err = ar.Read()
+	if err != nil {
+		return nil, err
+	}
+	defer ar.Close()
+
+	return ar, nil
 }
 
 func readArtifact(c *cli.Context) error {
 	if c.NArg() == 0 {
 		return errors.New("Nothing specified, nothing validated. \nMaybe you wanted" +
 			"to say 'artifacts validate <pathspec>'?")
 	}
-	parsers, err := read(c.Args().First())
+
+	r, err := read(c.Args().First())
 	if err != nil {
 		return err
 	}
 
-	fmt.Printf("Read Mender artifact file containing %d updates\n", len(parsers))
+	parsers := r.GetWorkers()
+	info := r.GetInfo()
 
-	for k, p := range parsers {
-		fmt.Printf("details of update: %s\n", k)
-		fmt.Printf("  update type: '%s'\n", p.GetUpdateType().Type)
-		fmt.Printf("  supported device: '%s'\n", p.GetDeviceType())
-		if ri, ok := p.(*parser.RootfsParser); ok {
-			fmt.Printf("  image id: '%s'\n", ri.GetImageID())
-		}
+	fmt.Printf("Mender artifact:\n")
+	fmt.Printf("  Name: %s\n", r.GetArtifactName())
+	fmt.Printf("  Format: %s\n", info.Format)
+	fmt.Printf("  Version: %d\n", info.Version)
+	fmt.Printf("  Compatible devices: '%s'\n", r.GetCompatibleDevices())
 
+	fmt.Printf("\nUpdates:\n")
```

Comment:
Ah, from a command line option. Yes, makes sense, I thought it was debug prints.